packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -601,6 +601,27 @@ export abstract class BaseService extends Resource
     return cloudmapService;
   }
 
+  /**
+   * Associates this service with a CloudMap service
+   */
+  public associateCloudMapService(options: AssociateCloudMapServiceOptions): void {
```

Comment:
@SoManyHs Hey. I saw you self-assigned this PR. I thought it might leave a note on my thinking here:

I created `associateCloudMapService` only because I wasn't sure how I could make `enableCloudMap` accept a `cloudmap.IService` via `CloudMapOptions`. To make a non-breaking change, `enableCloudMap` needs to return the concrete `cloudmap.Service` type. But, I felt it would be useful to allow the customer to provide an imported `cloudmap.IService`. So, I added another member function to work around this issue.

Another option might be to create some, `enableCloudMapV2` that returns `cloudmap.IService` and deprecate the old one. What are your thoughts?

Reply:
Hi @misterjoshua,

Oooh, yeah not having the `cloudmapService` field have an interface type was definitely a miss in our initial implementation. I like the approach you took here, since having an `enableCloudMapV2` method is a little clunky for sure.  The good news is that we are in the process of working towards a v2 of the ECS modules, so we can incorporate breaking changes like changing the method signature for `enableCloudMap`. 

Tangentially, I don't know if you've checked out [ECS Service Extensions](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk-containers/ecs-service-extensions), but the philosophy there is to treat components such as cloudmap (here, wrapped up in Appmesh) more atomically.

Reply:
> Hi @misterjoshua,
> 
> Oooh, yeah not having the `cloudmapService` field have an interface type was definitely a miss in our initial implementation. I like the approach you took here, since having an `enableCloudMapV2` method is a little clunky for sure. The good news is that we are in the process of working towards a v2 of the ECS modules, so we can incorporate breaking changes like changing the method signature for `enableCloudMap`.

@SoManyHs Sounds good! Looking forward to v2. :)

> Tangentially, I don't know if you've checked out [ECS Service Extensions](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk-containers/ecs-service-extensions), but the philosophy there is to treat components such as cloudmap (here, wrapped up in Appmesh) more atomically.

Ah yes, that's an excellent module. I had originally wanted this PR's capability to work around difficulties in adding the service name as an environment variable in the first container on a task definition. It was to get JGroups clustering working on ECS for an embedded distributed cache & KV store. Thankfully, after #13240 hit it isn't as tricky. And, with this PR, there will be a couple of good ways to go about this. :)